galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/routes/CPSRoute.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.framework.api.cps.internal.routes;
+
+import java.nio.file.DirectoryStream;
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import com.google.gson.Gson;
+
+import dev.galasa.framework.ResourceNameValidator;
+import dev.galasa.framework.api.common.BaseRoute;
+import dev.galasa.framework.api.common.ResponseBuilder;
+import dev.galasa.framework.spi.ConfigurationPropertyStoreException;
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.utils.GalasaGsonBuilder;
+
+
+/**
+ * An abstract route used by all the Run-related routes.
+ */
+public abstract class CPSRoute extends BaseRoute {
+
+    static final ResourceNameValidator nameValidator = new ResourceNameValidator();
+    static final Gson gson = GalasaGsonBuilder.build();
+
+    // Define a default filter to accept everything
+    static DirectoryStream.Filter<Path> defaultFilter = path -> { return true; };
```

Comment:
Don't see this being used anywhere - could remove if not needed (same goes for other routes).